lib/cc/engine/analyzers/violation.rb

```diff
@@ -45,7 +64,9 @@ def sorted_hashes
         end
 
         def other_sexps
-          @other_locations ||= sorted_hashes.drop(1)
+          sorted = sorted_hashes
```

Comment:
Doing a mutation on the array here is unfortunate. Can we do better? In fact, tracking the active hash by index up in `#initialize` feels like it could be cleaner.

We could track the actual current directly instead:

```
def initialize((language_strategy, issue, hashes, current_sexp = nil)
    ...
    @location = current_sexp || sorted_hashes.first
end

def other_sexps
  sorted_hashes.reject { |other| other == current_sexp }
end
```